openhcl/openhcl_boot/src/main.rs

```diff
@@ -331,6 +334,19 @@ fn reserved_memory_regions(
             ReservedMemoryType::Vtl2Reserved,
         ));
     }
+    let dma_4k_pages = partition_info.preserve_dma_4k_pages.unwrap_or(0);
+    // If DMA reserved hint was provided by Host, allocate top of VTL2 memory range
```

Comment:
This is actually insufficient - you need to report to sidecar allocation in `start_sidecar` that these ranges are not available for allocation. This fn is called after.

Reply:
Looking into this and have questions.
1. There are other reserved memory regions, none of which is passed to `start_sidecar` because we calculate those regions afterwards.
2. Sidecar itself is the reserved region from our point of view.

So how do we restrict sidecar from using today's VFIO DMA, for example? My assumption was that sidecar cannot access any memory beyond its assigned space (e.g. VFIO DMA is inaccessible), and that space is defined in IGVM. Do you mean that fixed_pool mem could be the part of sidecar range?

Reply:
it's a bit confusing today, because some of the regions are captured as part of the `ShimParams::used` range, this describes the whole range used by the loader. Those _existing_ reserved ranges got accounted for by that.

However, any _new_ ranges we allocate (like pool memory) needs to be accounted for, so when we load sidecar we tell it which ranges are legal for it to allocate from. I have on my list to clean all of this address space management in the bootshim up very soon, but you can see how I solved this in my WIP branch: https://github.com/chris-oo/openvmm/commit/33cb4eae47ca7379571263b70465398f16300f3e. What I've done here is add some new accounting to track _all_ ranges at allocation time, since we need to track all of it. 

I may solve this for you if I get the pool PR in first.

Reply:
To be closed after integration with Chris's changes.